tfx/components/example_gen/utils.py

```diff
@@ -276,19 +322,40 @@ def _retrieve_latest_span_version(
   logging.info('Regex pattern for split %s: %s', split.name,
                split_regex_pattern)
 
+  latest_span_elems = None
+  latest_span_int = None
+  latest_version = None
+
   files = tf.io.gfile.glob(split_glob_pattern)
 
   for file_path in files:
     result = re.search(split_regex_pattern, file_path)
     if result is None:
       raise ValueError('Glob pattern does not match regex pattern')
 
-    span_str = result.group(SPAN_PROPERTY_NAME)
-    try:
-      span_int = int(span_str)
-    except ValueError:
-      raise ValueError('Cannot find %s number from %s based on %s' %
-                       (SPAN_PROPERTY_NAME, file_path, split_regex_pattern))
+    span_strs = None
+    span_int = None
+
+    if is_match_span:
+      span_strs = [result.group(SPAN_PROPERTY_NAME)]
+      try:
+        span_int = int(span_strs[0])
+      except ValueError:
+        raise ValueError('Cannot find %s number from %s based on %s' %
+                         (SPAN_PROPERTY_NAME, file_path, split_regex_pattern))
+
+    elif is_match_date:
+      span_strs = [result.group(name) for name in ['year', 'month', 'day']]
```

Comment:
nit, for ['year', 'month', 'day'], using some constants?

Reply:
constraints are handled by the `datetime` package